src/hooks/useModifyCharacters.ts

```diff
@@ -2,44 +2,196 @@ import { useReducer } from 'react'
 
 import { ModifiedCharacter } from '../api/starWars-types'
 
+export const CHARACTER_ACTION_TYPES = {
+  SORT: 'sort',
+  FILTER: 'filter',
+} as const
+
 export const SORT_FACTOR = {
   MASS: 'mass',
   HEIGHT: 'height',
 } as const
 
+export const FILTER_FACTOR = {
+  HOMEWORLD: 'homeworld',
+  GENDER: 'gender',
+} as const
+
 // "mass" | "height"
 export type SortFactor = typeof SORT_FACTOR[keyof typeof SORT_FACTOR]
 
+export type FilterFactor = typeof FILTER_FACTOR[keyof typeof FILTER_FACTOR]
+
+type FilterTargets = {
+  homelandTarget: string
+  genderTarget: string
+}
+
 type CharacterState = {
   originalCharacters: ModifiedCharacter[]
-  modifiedCharacters: ModifiedCharacter[]
-  selectedValue: SortFactor | ''
+  renderedCharacters: ModifiedCharacter[]
+  sortFactor: SortFactor | ''
+  homeworldFilters: string[]
+  genderFilters: string[]
+  filterTargets: FilterTargets
 }
 
-type CharacterAction = { type: 'sort'; sortFactor: CharacterState['selectedValue'] }
+type CharacterAction =
+  | {
```

Comment:
What is the use of the first pipe, `|` ?

Reply:
It's a prettier thing. It doesn't add anything but it's surely annoying